subctl/issue-798-show-versions-from-logs.md

```diff
@@ -0,0 +1,106 @@
+# Extract version information from pod los
+
+Related issue:
+[show versions should extract version information from pod logs](https://github.com/submariner-io/subctl/issues/798)
+
+## Summary
+
+`show versions` currently shows version of the `DaemonSet/Deployment` which is the version configured. But actual version
+depends on the image running on the pods and can vary based on ImageOverrides, mirrors etc.
+
+This proposal describes a way to show actual version running on the pods.
+
+## Proposal
+
+Along with the version configured in the `DaemonSet/Deployment` and Submariner CR, `show version` will also show actual
+version running in the pods. All submariner components will log the version during bring up and `subctl` will extract
```

Comment:
Will the version log lines persist in long-running deployments?

Reply:
It won’t... That’s why I suggested also adding a `--version` flag.

Reply:
@skitt @dfarrell07 How about we use `--version` flag for `subctl show version` and skip looking at logs for it? Will make things simpler. We will still log the version, useful for gather output.

Reply:
I’m wondering if there might be scenarios where we’re not allowed to execute a binary in the existing pods, but perhaps we can worry about that later. One benefit I can see to looking at the logs is that we’d be able to do that on dead pods too, where we couldn’t execute commands anyway.

It would be fine by me to change this EP to only target the `--version` variant (while still logging for informational purposes, as you mention), and leave log parsing as a future enhancement if necessary (but mention it as such in the EP).

Reply:
How about, try to extract using `--version` and if it fails then try log? We can do both as part of this enhancement itself.

Reply:
That’s fine too, and if we can do both we can change the order if necessary.